flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -461,16 +461,15 @@ AbstractInvokable getInvokable() {
 		return invokable;
 	}
 
-	public StackTraceElement[] getStackTraceOfExecutingThread() {
-		final AbstractInvokable invokable = this.invokable;
-
-		if (invokable == null) {
-			return new StackTraceElement[0];
+	public boolean isAvailableForOutput() {
+		if (invokable == null || consumableNotifyingPartitionWriters.length == 0) {
+			return true;
 		}
-
-		return invokable.getExecutingThread()
-			.orElse(executingThread)
-			.getStackTrace();
+		final CompletableFuture<?>[] outputFutures = new CompletableFuture[consumableNotifyingPartitionWriters.length];
+		for(int i = 0; i < outputFutures.length; ++i) {
```

Comment:
whitespace after `for`

Reply:
In my another PR I already implemented the logic of checking output available in `StreamTask`, so maybe we could make `AbstractInvokable` implement `BackPressureSampleableTask` to reuse that logic. Then we could avoid creating the array every time.

Reply:
There is only an ```AbstractInvokable``` filed in ```Task```, not ```StreamTask```, if we reuse the code in ```StreamTask```, we may have to make a type cast and handle other type of ```AbstractInvokable``` except for ```StreamTask```. Maybe the current implementation without code reuse is more simple and clean.

Reply:
No need type cast. 

If `AbstractInvokable` implements the interface of `BackPressureSampleableTask`, while calling `Task#isBackPressured` -> `invokable#isBackPressured`. And in the `StreamTask#isBackPressured` implementation it would reuse the codes in my PR, for the `BatchTask#isBackPressured` it can always return false as now. 

The only concern is for `BatchTask#isBackPressured`, we make a consumption that batch job is always using blocking partition which would never be backpressured by downstream.

The current implementation would create the arrays every time, if the trigger is not very frequent, the overhead can also be accepted I guess.